### PR TITLE
Optimized and refactored oneDNN layer_norm kernel

### PR DESCRIPTION
**Pull request by jakpiase**
### PR types
Others

### PR changes
OPs

### Describe
Layernorm was copying scaleshift data two times, which was unnecessary. I have deleted that and I have also refactored this code. Mutable->data was also called two times, so I have removed it as well

### CODE REVIEW

**Comment by jakpiase** [comment]
@tsocha Please review that PR

File: paddle/fluid/operators/mkldnn/layer_norm_mkldnn_op.cc [hunk 1]
```diff
@@ -133,20 +132,7 @@ class LayerNormMKLDNNOpKernel : public paddle::framework::OpKernel<T> {
 
     std::shared_ptr<mkldnn::memory> scaleshift_memory;
```
Comment (lidanqing-vv) [inline paddle/fluid/operators/mkldnn/layer_norm_mkldnn_op.cc]: Can you put this `std::shared_ptr<mkldnn::memory> scaleshift_memory;` inside `if (with_scaleshit)` ?

File: paddle/fluid/operators/mkldnn/layer_norm_mkldnn_op.cc [hunk 2]
```diff
@@ -133,20 +132,7 @@ class LayerNormMKLDNNOpKernel : public paddle::framework::OpKernel<T> {
...
-      // data
-      std::vector<float> scaleshift_data;
-      scaleshift_data.reserve(2 * C);
-      scaleshift_data.insert(scaleshift_data.begin(), scale->data<float>(),
-                             scale->data<float>() + C);
-
-      scaleshift_data.insert(scaleshift_data.end(), bias->data<float>(),
-                             bias->data<float>() + C);
-
-      scaleshift_memory = handler.AcquireScaleShiftMemory(scaleshift_data);
```
Comment (lidanqing-vv) [inline paddle/fluid/operators/mkldnn/layer_norm_mkldnn_op.cc:149]: I saw 
```
// TODO(jczaja): Enable FP32 when performance is good
namespace ops = paddle::operators;
REGISTER_OP_KERNEL(layer_norm, MKLDNN, ::paddle::platform::CPUPlace,
                   ops::LayerNormMKLDNNOpKernel<paddle::platform::bfloat16>);
```
in the last few lines of this file. So does this PR improve the performance and hence could register FP32 kernel layer_norm ?

File: paddle/fluid/operators/mkldnn/layer_norm_mkldnn_op.cc (hunk 1 above)
Comment (jakpiase) [reply paddle/fluid/operators/mkldnn/layer_norm_mkldnn_op.cc]: Ok

File: paddle/fluid/operators/mkldnn/layer_norm_mkldnn_op.cc (hunk 2 above)
Comment (jakpiase) [reply paddle/fluid/operators/mkldnn/layer_norm_mkldnn_op.cc:149]: Unfortunately not. I have deleted one unnecessary copying of scaleshift tensor, but that have improved the performance by around ~1%, so this PR is more like refactoring. Our layernorm is still almost 2x slower than native kernel and that will need further investigation

**Review by jczaja — APPROVED** [review]
LGTM, you could generate flamegraph (without LN vs with LN) to see if any problem is revealed. And if not then maybe run Top down microarchitectue analysis to see where is ti bound. Anyway nice PR.

**Review by tsocha — APPROVED** [review]

**Comment by jakpiase** [comment]
@jczaja please merge that PR

**Pull request merged**